test/e2e/storage/drivers/in_tree.go

```diff
@@ -1367,9 +1367,11 @@ func (g *gcePdDriver) PrepareTest(f *framework.Framework) (*storageframework.Per
 
 func (g *gcePdDriver) CreateVolume(config *storageframework.PerTestConfig, volType storageframework.TestVolType) storageframework.TestVolume {
 	zone := getInlineVolumeZone(config.Framework)
-	if volType == storageframework.InlineVolume {
-		// PD will be created in framework.TestContext.CloudConfig.Zone zone,
-		// so pods should be also scheduled there.
+	if zone == "" {
+		framework.Fail("could not determine suitable zone for volume")
```

Comment:
Are you sure that there's never a case where this test currently runs without a zone? This additional check would break that case.